### PR TITLE
Add test-plan skill

### PR DESCRIPTION
**Pull request by haacked**
## Summary

- Adds a Claude Code skill (`/test-plan`) that generates a structured GitHub Flavored Markdown test plan checklist
- Supports two input modes: **branch mode** (analyzes the current branch's diff against base) and **plan mode** (analyzes an implementation plan file via `--plan <file>`)
- Produces detailed, actionable test items grouped into numbered sections, with emphasis on concrete identifiers, negative assertions, and integration boundary coverage

## Test plan

- [ ] Run `/test-plan` on a branch with changes — verify it generates a checklist with specific, actionable items
- [ ] Run `/test-plan --plan <file>` with an implementation plan — verify it skips git analysis and generates from the plan
- [ ] Run `/test-plan --force` — verify it skips the preview and saves immediately
- [ ] Run `/test-plan --save /tmp/out.md` — verify it saves to the custom path
- [ ] Run `/test-plan` on a branch with no changes — verify it reports "no changes" and stops

### CODE REVIEW

File: ai/skills/test-plan/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+**Section count and density:** Aim for one section per distinct behavior cluster (cache population, each invalidation trigger, validation logic, consistency checks). Each section should have 4–10 items. A complete test plan typically has 6–10 sections. Fewer than 4 sections or fewer than 20 total items usually means you have not gone deep enough.
+
+**Item quality standard:** Every item must name the concrete thing being tested. Compare:
+
+| Too vague — do not write this | Specific enough — write this instead |
+|-------------------------------|--------------------------------------|
+| Verify cache invalidation works after token rotation | Rotating a team's secret API token invalidates the discarded backup token's cache entry in Redis; the old primary (now backup) token's entry is NOT deleted |
+| Test that scoped PAKs are rejected correctly | PAK with `scopes: ["session_recording:read"]` (no feature flag scope) is rejected when requesting flags |
+| Check that creating a new key doesn't cause issues | Creating a new PAK does NOT trigger any cache invalidation (nothing cached yet) |
+
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/skills/test-plan/SKILL.md:118]: The table is already a valid 2-column GFM table. Each row starts with a single `|` — there is no double `||`. Lines 113–117 of the file confirm this. The diff prefix `+` at the start of each diff line is not part of the file content.

File: ai/skills/test-plan/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+Extract and record these concrete details from the source material — you will embed them verbatim in test items:
+
+- **Literal identifiers**: constant names, cache key patterns (e.g., `posthog:auth_token:{sha256_hash}`), token prefixes (e.g., `phs_`, `phx_`), field names (e.g., `scoped_teams`, `org_ids`), enum values, signal names
+- **Boundaries and integration points**: every place the code crosses a system boundary (API endpoint → database, Python → Redis → Rust, signal handler → async task, model save → cache invalidation)
+- **State transitions that trigger behavior**: which model fields, when saved or deleted, cause which downstream effects — and which do NOT
+- **Conditional logic**: branches where one path causes an effect and the other does not; these generate pairs of positive and negative test items
+- **Security-sensitive paths**: authentication checks, permission scoping, token rotation, access denial
+- **Assumptions that could silently break**: hash function compatibility between components, key format consistency, flag/feature parity between services
+
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/skills/test-plan/SKILL.md:79]: The "embed verbatim" instruction refers to code identifiers extracted from git diffs and plan files — constant names, cache key patterns (as templates like `posthog:auth_token:{sha256_hash}`), field names, enum values. These are structural code artifacts, not runtime secret values. Actual secrets should not appear in committed diffs or plan documents in a well-managed project, and the examples in the skill make the intent clear. Adding a generic disclaimer for every possible misuse would add noise without meaningfully improving the skill.

File: ai/skills/test-plan/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+### 1. Parse Arguments
+
+Extract from user input:
+
+- `force` = true if `--force` is present
+- `plan_file` = path after `--plan`, or empty (branch mode)
+- `save_path` = path after `--save`, or `.context/test-plan.md`
+
+If `--plan` is provided, use plan mode. Otherwise, use branch mode.
+
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/skills/test-plan/SKILL.md:39]: This is an LLM skill, not a shell script. The argument parsing instructions are interpreted by an AI agent, which handles edge cases like `--plan` without a following value naturally and intelligently — it will ask the user or stop with an error. Over-specifying every validation branch in an LLM prompt adds noise without value. Step 2 already covers the most important validation ("If the file does not exist, tell the user and stop"), which is the failure mode that actually matters at runtime.

File: ai/skills/test-plan/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+gh repo view --json defaultBranchRef -q .defaultBranchRef.name
+```
+
+If `gh` fails (no remote, not a GitHub repo), fall back to `main` if it exists, otherwise `master`.
+
+Then run in parallel:
+
+```bash
+git log <base>..HEAD --oneline
+git diff <base>..HEAD
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/skills/test-plan/SKILL.md]: Good catch — updated to prefer git-native remote HEAD detection, fall back to `gh` only if available, then `main`/`master` only if they exist locally, and stop with a clear message if nothing resolves.

File: ai/skills/test-plan/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+## Arguments
+
+- `--force` — skip preview and confirmation; save immediately
+- `--plan <file>` — path to an implementation plan file (skip git analysis, use this file as input)
+- `--save <path>` — custom output path (default: `.context/test-plan.md`)
+
+Example invocations:
+
+- `/test-plan` — generate from current branch diff
+- `/test-plan --plan ~/dev/ai/plans/posthog/posthog/auth-cache.md` — generate from plan file
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/skills/test-plan/SKILL.md]: Fixed — switched to a generic example path (`./plans/my-feature.md`).

File: ai/skills/test-plan/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+   ```
+
+4. If `<base>` is still empty, tell the user you cannot determine a base branch and **stop**.
+
+Then run in parallel:
+
+```bash
+git log <base>..HEAD --oneline
+git diff <base>..HEAD
+```
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/skills/test-plan/SKILL.md]: Fixed — updated the git log/diff commands to use `"$base"` consistently with the detection snippets above.

File: .notes/reviews/test-plan-skill.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+\```
+```
+
+This saves the full diff fetch in the no-changes case. However, this trades parallelism for short-circuiting. For the common case (branch has changes), the current approach of fetching both in parallel would be faster. A pragmatic middle ground: run both in parallel and add the short-circuit as-is, accepting the wasted diff in the rare no-changes case. In that case, just add "run in parallel" to the second batch per issue #1.
+
+**Confidence:** 80
+
+## Summary
+
+The main actionable finding is that the second batch of git commands (`git log` + `git diff`) lacks an explicit "run in parallel" instruction, unlike the equivalent pattern in the `create-pr` skill. Adding those two words will save a round-trip on every invocation. The unused `git rev-parse` is a minor cleanup. The short-circuit ordering is a design tradeoff that may not be worth changing depending on how often the skill is invoked on branches with no changes.
```
[the remark this answers is not in the document]
Comment (haacked) [reply .notes/reviews/test-plan-skill.md]: Agreed — the review note was stale after the base-branch detection rewrite. Deleted it.

File: ai/commands/squash.md [hunk 8]
```diff
@@ -25,8 +25,8 @@ A **CI snapshot commit** is any commit whose author email contains `[bot]` (e.g.
 ### 1. Gather Context
 
 ```bash
-BASE_BRANCH=$(gh repo view --json defaultBranchRef -q .defaultBranchRef.name)
-MERGE_BASE=$(git merge-base HEAD "$BASE_BRANCH")
+BASE_BRANCH=$(bash "$HOME/.dotfiles/bin/lib/git-default-branch.sh")
+MERGE_BASE=$(git merge-base HEAD "origin/$BASE_BRANCH")
```
[the remark this answers is not in the document]
Comment (haacked) [reply ai/commands/squash.md:29]: This file is an AI instruction document, not a shell script — the code blocks are pseudocode patterns that Claude executes step by step, so adding bash error-handling boilerplate would make the prompt verbose and harder to follow. The correct pattern for these skill files is a natural language stop condition (matching the `test-plan/SKILL.md` convention). Added: "If the helper is not available or `BASE_BRANCH` is empty, tell the user and **stop**."
